### PR TITLE
Block process execution with seccomp on linux/amd64

### PR DESCRIPTION
**Pull request by rmuir**
On newer linux kernels, we can use prctl/seccomp to lock down the process and prevent the worst of the worst, like execution. This is used by e.g. chrome/firefox sandbox and so on.

This is just another level of security, java's security manager is not perfect, and there are often bugs in java itself, so it would good to have another level of defense.

See https://en.wikipedia.org/wiki/Seccomp for more information

This PR blocks execve(), fork(), and vfork(), returning EACCES instead, so even with security manager disabled, process execution is still prevented. 

```
            java.io.IOException: Cannot run program "ls": error=13, Permission denied
                    at __randomizedtesting.SeedInfo.seed([65E6C4BED11899E:FC6E1CA6AA2DB634]:0)
                    at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)
                    at java.lang.Runtime.exec(Runtime.java:620)
                    ...
                  Caused by: java.io.IOException: error=13, Permission denied
                    at java.lang.UNIXProcess.forkAndExec(Native Method)
                    at java.lang.UNIXProcess.<init>(UNIXProcess.java:248)
                    at java.lang.ProcessImpl.start(ProcessImpl.java:134)
                    at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)
```

### CODE REVIEW

**Comment by s1monw** [comment]
nice!

**Comment by dakrone** [comment]
Tested this locally and it looks to be working, very neat!

**Comment by rmuir** [comment]
I fixed all the TODOs and gripes I had with this and added documentation, checks here. I think its ready...

File: core/src/main/java/org/elasticsearch/bootstrap/JNANatives.java [hunk 1]
```diff
@@ -170,4 +172,16 @@ static void addConsoleCtrlHandler(ConsoleCtrlHandler handler) {
...
 
+    static void trySeccomp() {
+        if (Constants.LINUX && "amd64".equals(Constants.OS_ARCH)) {
+            try {
+                Seccomp.installFilter();
+                LOCAL_SECCOMP = true;
+            } catch (Exception e) {
+                // this is likely to happen unless the kernel is newish, its a best effort at the moment
+                // so unfortunately, we hide the stacktrace for now...
+                logger.warn("unable to install seccomp filter: " + e.getMessage());
```
Comment (jaymode) [inline core/src/main/java/org/elasticsearch/bootstrap/JNANatives.java]: What do you think about `DEBUG` or `TRACE` log with the exception and if DEBUG/TRACE isn't enabled we just log what is there now? I remember a stack trace from one of these helped in the past

**Comment by jaymode** [comment]
left a minor comment about a log message, other than that LGTM! This is really nice.

**Comment by rmuir** [comment]
Thanks for looking, I added that debug, good idea.

**Comment by s1monw** [comment]
LGTM FWIW I don't know that linux feature in particular

**Comment by samcday** [comment]
Hey so it turns out this secure computing thing completely breaks `-XX:OnOutOfMemoryError`, since the JVM attempts to `execs`the supplied command(s).

Should I raise a separate issue for that?

**Comment by rmuir** [comment]
More like, launching processes is bullshit for a daemon process to do. Under any circumstances. Manage this stuff with startup scripts, etc. If you don't agree with me, open an issue if you like. But this is an important piece, to ensure remote execution is something that doesn't happen again.

**Comment by jasontedor** [comment]
> Hey so it turns out this secure computing thing completely breaks `-XX:OnOutOfMemoryError`, since the JVM attempts to `execs`the supplied command(s).

For others that might not click through to #18736, if all you want to do is kill on `OutOfMemoryError`, just use `ExitOnOutOfMemoryError` available starting in 8u92.

**Pull request closed without merging**